data_structures/queue/circular_queue.py

```diff
@@ -0,0 +1,41 @@
+# Implementation of Circular Queue (using python lists)
+
+class CircularQueue:
+    """ Circular queue with fixed capacity """
+
+    def __init__(self, n):
+        self.n = n
+        self.array = [None]*self.n
+        self.front = 0              # For first ele. Index
+        self.rear = 0
+        self.size = 0
+
+    def __len__(self):
+        return self.size
+
+    def is_empty(self):
+        return self.size == 0
+
+    def first(self):
+        if self.is_empty():
+            return False
+        return self.array[self.front]
+
+    def enqueue(self, data):
+        if self.size >= self.n:
+            raise "QUEUE IS FULL"
```

Comment:
Raising a string might have worked in Python 2 but it does not in Python 3.

Reply:
Thank you, sir,  Yeah sure I'll remove it but i guess instead of raising a string, returning True or False might be a better option

Reply:
Please consider `raise Exception("Queue is full")`

Reply:
Raising an exception is more Pythonic.